diff --git ts/private/ts_proto_library.bzl ts/private/ts_proto_library.bzl
index 9866f8f..223c1f4 100644
--- ts/private/ts_proto_library.bzl
+++ ts/private/ts_proto_library.bzl
@@ -50,7 +50,7 @@ def _protoc_action(ctx, proto_info, outputs, options = {
         args.add_joined(["--connect-query_out", ctx.bin_dir.path], join_with = "=")
 
     args.add("--descriptor_set_in")
-    args.add_joined(proto_info.transitive_descriptor_sets, join_with = ":")
+    args.add_joined(proto_info.transitive_descriptor_sets, join_with = ctx.configuration.host_path_separator)
 
     args.add_all(proto_info.direct_sources)
 
