diff --git npm/private/npm_package_store.bzl npm/private/npm_package_store.bzl
index a794c201..c8055978 100644
--- npm/private/npm_package_store.bzl
+++ npm/private/npm_package_store.bzl
@@ -1,6 +1,7 @@
 "npm_package_store rule"
 
 load("@aspect_bazel_lib//lib:copy_directory.bzl", "copy_directory_bin_action")
+load("@aspect_bazel_lib//lib:platform_utils.bzl", "platform_utils")
 
 # buildifier: disable=bzl-visibility
 load("//js/private:js_info.bzl", "JsInfo", "js_info")
@@ -225,12 +226,16 @@ def _npm_package_store_impl(ctx):
                 # `chmod -R a+X` to fix up these packages (https://stackoverflow.com/a/14634721).
                 # See https://github.com/aspect-build/rules_js/issues/1637 for more info.
                 bsdtar = ctx.toolchains["@aspect_bazel_lib//lib:tar_toolchain_type"]
+                host_is_windows = platform_utils.host_platform_is_windows()
                 args = ctx.actions.args()
-                args.add(bsdtar.tarinfo.binary)
+                if host_is_windows:
+                    args.add("bsdtar")
+                else:
+                    args.add(bsdtar.tarinfo.binary)
                 args.add(src)
                 args.add(package_store_directory.path)  # Need to use `.path` due to: Error in add: Cannot add directories to Args#add since they may expand to multiple values. Either use Args#add_all (if you want expansion) or args.add(directory.path).
                 ctx.actions.run_shell(
-                    tools = [bsdtar.tarinfo.binary],
+                    tools = [] if host_is_windows else [bsdtar.tarinfo.binary],
                     inputs = depset(direct = [src], transitive = [bsdtar.default.files]),
                     outputs = [package_store_directory],
                     command = "$1 --extract --no-same-owner --no-same-permissions --strip-components 1 --file $2 --directory $3 && chmod -R a+X $3",
